plugins/inputs/statsd/datadog_test.go

```diff
@@ -0,0 +1,472 @@
+package statsd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventGather(t *testing.T) {
+	now := time.Now()
+	type expected struct {
+		title  string
+		tags   map[string]string
+		fields map[string]interface{}
+	}
+	tests := []struct {
+		name     string
+		message  string
+		hostname string
+		now      time.Time
+		err      bool
+		expected expected
+	}{{
+		name:     "basic",
+		message:  "_e{10,9}:test title|test text",
+		hostname: "default-hostname",
+		now:      now,
+		err:      false,
+		expected: expected{
+			title: "test title",
+			tags:  map[string]string{"source": "default-hostname"},
+			fields: map[string]interface{}{
+				"priority":   priorityNormal,
+				"alert_type": "info",
+				"text":       "test text",
+			},
+		},
+	},
+		{
+			name:     "escape some stuff",
+			message:  "_e{10,24}:test title|test\\line1\\nline2\\nline3",
+			hostname: "default-hostname",
+			now:      now.Add(1),
+			err:      false,
+			expected: expected{
+				title: "test title",
+				tags:  map[string]string{"source": "default-hostname"},
+				fields: map[string]interface{}{
+					"priority":   priorityNormal,
+					"alert_type": "info",
+					"text":       "test\\line1\nline2\nline3",
+				},
+			},
+		},
+		{
+			name:     "custom time",
+			message:  "_e{10,9}:test title|test text|d:21",
+			hostname: "default-hostname",
+			now:      now.Add(2),
+			err:      false,
+			expected: expected{
+				title: "test title",
+				tags:  map[string]string{"source": "default-hostname"},
+				fields: map[string]interface{}{
+					"priority":   priorityNormal,
+					"alert_type": "info",
+					"text":       "test text",
+					"ts":         int64(21),
+				},
+			},
+		},
+	}
+	acc := &testutil.Accumulator{}
+	s := NewTestStatsd()
+	s.acc = acc
+
+	for i := range tests {
+		t.Run(tests[i].name, func(t *testing.T) {
+			err := s.parseEventMessage(tests[i].now, tests[i].message, tests[i].hostname)
+			if tests[i].err {
+				require.NotNil(t, err)
+			} else {
+				require.Nil(t, err)
+			}
+			require.Equal(t, uint64(i+1), acc.NMetrics())
+
+			require.Nil(t, err)
+			require.Equal(t, tests[i].expected.title, acc.Metrics[i].Measurement)
```

Comment:
These are fine, no need to update, but in the future use this method:
https://github.com/influxdata/telegraf/blob/master/plugins/inputs/minecraft/minecraft_test.go#L89
https://github.com/influxdata/telegraf/blob/master/plugins/inputs/minecraft/minecraft_test.go#L121